### PR TITLE
refactor: weapon list page ft. css grid

### PR DESCRIPTION
**Pull request by vikarina**
Created component cards for weapon summaries and simplified the DOM structure, applying the same logic / stylings as on the character list page. Demo image attached below:

weapons-page-rework

### CODE REVIEW

File: src/app/components/weapon-list/weapon-list.component.html [hunk 1]
```diff
@@ -1,75 +1,24 @@
...
+    <div class="weapon-filter-sort-stub"></div>
+    <div class="weapon-group-section" *ngFor="let listItem of listItems | keyvalue">
+      <div class="weapon-group-section-header">
+        <img *ngIf="listItem.value.mainImage" src="{{ listItem.value.mainImage }}" />
+        <h3 *ngIf="listItem.value.mainLabel">
+          {{ listItem.value.mainLabel }}
+        </h3>
+      </div>
+      <div class="weapon-items-list">
+        <moe-weapon-card *ngFor="let weapon of listItem.value.weapons"
```
Comment (Steborino) [inline src/app/components/weapon-list/weapon-list.component.html:18]: https://github.com/angular/angular/issues/9209 😪

**Review by Steborino — APPROVED** [review]
Mmm looks so nice 😎

File: src/app/components/weapon-list/weapon-list.component.html (hunk 1 above)
Comment (vikarina) [reply src/app/components/weapon-list/weapon-list.component.html:18]: this is the nerdiest thing I've ever read

Comment (Steborino) [reply src/app/components/weapon-list/weapon-list.component.html:18]: 🥓🤳

File: src/app/components/weapon-card/weapon-card.component.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      <div class="weapon-substat-label">{{ weapon.substatType }}</div>
+      <div class="weapon-substat-value">{{ weapon.substatValue }}</div>
+    </div>
+
+    <moe-rarity *ngIf="weapon.rarity > 0" [value]="weapon.rarity"></moe-rarity>
+  </div>
+
+  <div class="weapon-card-image-container"
+    [ngClass]="{
+      'rarity1-gradient': !weapon.rarity || weapon.rarity === 1,
```
[the remark this answers is not in the document]
Comment (vikarina) [reply src/app/components/weapon-card/weapon-card.component.html:22]: Is this ... readability ??

**Pull request merged**